apps/wallet/src/ui/app/hooks/useGetCoins.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useQuery } from '@tanstack/react-query';
+
+import { api } from '../redux/store/thunk-extras';
+
+import type { SuiAddress, PaginatedCoins, CoinStruct } from '@mysten/sui.js';
+// minimum number of coins to fetch per request
+const MIN_COINS_PER_REQUEST = 100;
+
+async function fetchCoins(address: SuiAddress, coinType: string) {
+    const rpc = api.instance.fullNode;
```

Comment:
I merged https://github.com/MystenLabs/sui/pull/8632 earlier today, so we should use `useRpcClient` from `@mysten/core` now instead of directly accessing `api.instance.fullNode` here